### ISSUE TITLE
issues on 1.6.1 with hercules mk2

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#198]
Reported by: egrenados
Date: 2008-12-09T15:18:06Z
Status: Won't Fix
Importance: Undecided
Launchpad Issue: [lp306558](https://bugs.launchpad.net/bugs/306558)

    Hello,

    I got my hercules mk2 1 week ago and tried it with mixxx 1.6.1 on linux/ubuntu (compiled with libDJconsole).
    I quickly ran into various issues that i fixed myself in the sources :

    1/ The headphone button is not well mapped : on hercules mk2 the button values for the different headphone mode should be :
    100 : trackA
    101 : track B
    102 : mix
    103 : split
    but these buttons does not seems to be defined in libDJconsole and are overwritten to different values in herculeslinux.cpp (51,52...)
    This lead into :
    -no sound in headphone for trackA / trackB 
    -mix is fine because it comes from the direct output of mixxx (no impact of 'headphone' buttons)
    -split have direct output (mix) on one side and nothing on the other
    by redefining the button numbers it works well (but maybe not for other hercules devices)

    2/ No headphone button is selected in mixxx when selecting 'split' output on the mk2. This is due to the herculeslinux.cpp code, muting both output. 
    Is there any reason for this ? I have changed my local code to select headphone on side opposite to crossfader position : i hear the mix on one ear 
    and the next song on other ear.

    3/ When changing pitch by using the beatsync/autobeat button, the old pitch value used in herculeslinux.cpp is not updated : if i want to move slightly the pitch from here using the pitch/rate button, the value come back to the previous position before pressing autobeat : very annoying !
    I have fixed it by reading the mixxx pitch value back into the hercules old pitch value before changing it and it works perfectly

    4/ Pitch bend mk2 button are mapped to permanent pitch changes. In general the pitch bend is used to brake/speedup the track temporarily. The mapping is done in hercules.cpp 'Standard mapping'. I have changed it to temporary pitch change button successfully

    5/ Scratch is really not looking like scratch : the speed does not seems to be increase when speeding up the jog: looks like going reverse and play always with the same speed :-( Not fixed yet

    I can provide my sources (or changes done) if required.

    Regards.

    m4nu.

**Comment by Holzhaus** [comment]
Created by: eythian
Date: 2008-12-09T20:42:28Z

    If you're willing to hold off for a little while, we're just waiting on Hercules to finalise some drivers, and the Mk2 (along with all the Herc consoles) will be a MIDI device on Linux. Not sure when this'll happen exactly, but I'm hoping quite soon.

**Comment by Holzhaus** [comment]
Created by: ironstorm-gmail
Date: 2008-12-09T21:56:43Z

    As per Eythian's comment, we will actually be removing libDJConsole from Mixxx in favour of a MIDI mapping solution based on an as yet unreleased driver from Hercules.    

    So because of that these bits won't be fixed by the dev team (that's not to say we won't accept patches into SVN for that code until Herc puts out the driver if you or someone else wants to write a patch to fix it - herculeslinux.cpp is the file that contains that block of code).

**Comment by Holzhaus** [comment]
Issue closed with status "Won't Fix".
